src/pybind/mgr/dashboard_v2/controllers/erasure_code_profile.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+
+from ..services.ceph_service import CephService
+from .. import logger, mgr
+from ..tools import ApiController, AuthRequired, RESTController
+
+
+@ApiController('erasure_code_profile')
+@AuthRequired()
+class ErasureCodeProfile(RESTController):
+
+    def list(self):
+        ret = []
+        for name, ecp in mgr.get('osd_map').get('erasure_code_profiles', {}).items():
+            ecp['name'] = name
+            ret.append(ecp)
+        return ret
+
+    @RESTController.args_from_json
+    def create(self, name, k, m, ruleset_failure_domain=None):
+        profile = ['k={}'.format(k), 'm={}'.format(m)]
+        if ruleset_failure_domain:
+            profile.append('crush-failure-domain={}'.format(ruleset_failure_domain))
+
+        CephService.send_command('mon', 'osd erasure-code-profile set', name=name,
```

Comment:
how is the potential errors of this call being handled?

Reply:
Error handling will change as soon as the task manager form @rjfd is merged. Right now, an exception is thrown and cached by the general exception handler which will render a JSON error containing an error message.